### PR TITLE
Fix off by one issue in event processing

### PR DESCRIPTION
**Pull request by nepoche**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #73 

Events are acted upon twice in the handle_event due to an off-by-one error when querying for a block range..

## What is the new behavior?

Events are only acted upon once.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by nepoche** [comment]
While running the tests for this PR, I came across an error which seems strange: 
After the second deposit, there is a ~30 second gap between signaling the bridge to create a proposal, and actually sending the transaction. These are the logs:
[signalling-proposal-gap.txt](https://github.com/webb-tools/relayer/files/7849223/signalling-proposal-gap.txt)

**Comment by drewstone** [comment]
@nepoche can you start using Gists instead of dumping txt files?

File: tests/integration_tests/ganache/anchorTests.ts [hunk 1]
```diff
@@ -301,11 +302,22 @@ describe('Anchor Tests', function () {
...
 
       // deposit
-      const deposit = await srcAnchor.deposit(chainId2);
+      console.log('before the first deposit');
+      await srcAnchor.deposit(chainId2);
 
       // allow time for the bridge proposal and execution
       console.log('waiting for bridge proposal and execution');
-      await sleep(30_000);
+      await sleep(40_000);
```
Comment (drewstone) [inline tests/integration_tests/ganache/anchorTests.ts:310]: Truly this type of addition to the tests does not make it more resilient or allow us to build resiliency into the relayer. In fact, I think it accomplishes the opposite. We need testing tools that "wait for bridge proposal and execution" by doing just that, not by waiting 40 seconds.

**Review by drewstone — APPROVED** [review]
LGTM for now. Can you please create tasks to improve the testing flow with utilities to properly wait for actions rather than sleeping for hardcoded times?

**Comment by shekohex** [comment]
Could you change these lines here https://github.com/webb-tools/relayer/blob/main/src/events_watcher/bridge_watcher.rs#L401-L411 by removing the `.block(...)` condition, and see if this make it faster? While this change is much more safer as in one block confirmation is not that bad, but this could be the behavior from ganache itself.

 however since ganache does not mine blocks on itself (there is must a transaction or something to force the chain to mine blocks).

I would suggest improving our tests by forcing ganache to mine a block every 1s for example, or just send `eth_mine` call to the chain to make it progress manually.

**Pull request merged**